docs/tutorials/add-oauth2-provider.md

```diff
@@ -1,56 +1,178 @@
+
+```shell
+$ git clone COPIED_URL
+```
+
+> To fork a repository, you will need a basic understanding of CLI and git-cli binaries installed. If you are a beginner, we recommend you to use `Github Desktop`. It is a really clean and simple visual Git client.
+
+Finally, you will need to create a `feat-XXX-YYY-oauth` branch based on the `master` branch and switch to it. The `XXX` should represent the issue ID and `YYY` the OAuth provider name.
+
+## 2. Implement new provider
+
+### 2.1 List your new provider
 
 The first step in adding a new OAuth2 provider is to add it to the list in providers config file array, located at:
 
 ```
-./app/config/providers.php
+app/config/providers.php
 ```
 
-Make sure to fill all data needed and that your provider array key name:
+Make sure to fill in all data needed and that your provider array key name:
 
-- is in camelCase format 
-- has no spaces or special characters.
+- is in [camelCase](https://en.wikipedia.org/wiki/Camel_case) format 
+- has no spaces or special characters
 
-### Add Provider Logo
+>  Please make sure to keep the list of providers in `providers.php` in the alphabetical order A-Z.
 
-Add a logo image to your new provider in this path: `./public/images/users`. Your logo should be a png 100×100px file with the name of your provider (all lowercase). Please make sure to leave about 30px padding around the logo to be consistent with other logos.
+### 2.2 Add Provider Logo
 
-### Add Provider Class
+Add a logo image to your new provider in this path: `public/images/users`. Your logo should be a png 100×100px file with the name of your provider (all lowercase). Please make sure to leave about 30px padding around the logo to be consistent with other logos.
 
-Once you have finished setting up all the metadata for the new provider, you need to start coding.
+### 2.3 Add Provider Class
 
-Create a new class that extends the basic OAuth2 provider abstract class in this location:
+Once you have finished setting up all the metadata for the new provider, you need to start coding.
 
-```bash
-./src/Auth/OAuth/ProviderName
+Create a new file `XXX.php` where `XXX` is the name of the OAuth provider in `CamelCase` with the first letter in upper case in this location
+"`bash
+src/Appwrite/Auth/OAuth
 ```
 
-Note that the class name should start with a capital letter as PHP FIG standards suggest.
+Inside this file, create a new class that extends the basic OAuth2 provider abstract class. Note that the class name should start with a capital letter, as PHP FIG standards suggest.
+
+Once a new class is created, you can start to implement your new provider's login flow. We have prepared a starting point for Oauth provider class below, but you should also consider looking at other provider's implementation and try to follow the same standards.
+
+```injectablephp
```

Comment:
```suggestion
```injectablephp
```
What's this :sweat_smile:

Reply:
No idea, PHPStorm autocompleted this, I didn't find just `php` 😅 I can see `php` in other places of Appwrite, let's use that.